scripts/checkFilesChanged.ts

```diff
@@ -0,0 +1,31 @@
+import 'zx/globals';
+import { ErrorHandler } from './util/errorHandling.js';
+
+const latestTag = await $`git --no-pager describe --tags $(git rev-list --tags --max-count=1)`;
+
+const latestTagCommitSha = await $`git --no-pager rev-list -n 1 ${latestTag.stdout.trim()}`;
+const diff = await $`git --no-pager diff --name-status ${latestTagCommitSha.stdout.trim()} HEAD schemas`;
+
+const gitDiffRegex = /^(?<action>A|M|D)\s+(?<file>schemas\/.+\.schema\.(?:json|ts))/;
+
+const addedAndModifiedFiles = diff.stdout
+  .split('\n')
+  .filter((line) => gitDiffRegex.test(line))
+  .map((line) => line.split('\t'));
+
+if (addedAndModifiedFiles.length === 0) {
+  console.log('No schema files have been added or modified since the latest release.');
+  process.exit(0);
+}
+const errorHandler = new ErrorHandler();
+errorHandler.addError(
+  ...addedAndModifiedFiles.map(([action, file]) => ({
+    file,
+    error: `Schema files should not be ${action === 'A' ? 'added' : 'modified'} outside of a release.`,
```

Comment:
What about deleted? You consider that a modification?

Reply:
fixed